src/mumble/AudioOutput.cpp

```diff
@@ -588,29 +588,42 @@ bool AudioOutput::mix(void *outbuff, unsigned int frameCount) {
 						aop->pfVolume[s] = -1.0;
 				}
 
+				if (!aop->piOffset) {
+					aop->piOffset = new unsigned int[nchan];
+					for (unsigned int s = 0; s < nchan; ++s)
+						aop->piOffset[s] = 0;
+				}
+
 				for (unsigned int s = 0; s < nchan; ++s) {
-					const float dot = bSpeakerPositional[s]
-										  ? connectionVec.x * speaker[s * 3 + 0] + connectionVec.y * speaker[s * 3 + 1]
-												+ connectionVec.z * speaker[s * 3 + 2]
-										  : 1.0f;
-					const float str   = svol[s] * calcGain(dot, len) * volumeAdjustment;
+					const float dot   = bSpeakerPositional[s]
+											? connectionVec.x * speaker[s * 3 + 0] + connectionVec.y * speaker[s * 3 + 1]
+                                                + connectionVec.z * speaker[s * 3 + 2]
+											: 1.0f;
+					const float str   = svol[s] * (0.05 + 19 * calcGain(dot, len) / 40.0) * volumeAdjustment;
```

Comment:
What is this doing and where are all these magic numbers coming from?

Reply:
`(0.05 + 19 * calcGain(dot, len) / 40.0)`

The volume is scaled to 19/20x and 0.05 is added. This will make the attenuation on the ear pointing away from the sound 0.05x instead of 0x (see graph below). I came to the value of 0.05 by looking at waveforms of ASMR/binaural audio recordings and trying to match the attenuation there. 

 realize it should be `(0.05 + 19 * calcGain(dot, len) / 20.0)` instead because the `calcGain()` function already divides the dot product by 2.

![attenuation-new](https://user-images.githubusercontent.com/38155544/120616071-708e5e80-c448-11eb-8571-61ad0005cdce.png)

Reply:
The addition of `0.05` is probably not needed as inside `calcGain` we have a feature called "Blooming" that makes sure that the volume on the other ear is not zero (if the sound source is close enough).

How did you arrive at the 19/20?

Reply:
The 19/20 comes from the 0.05, so that the gain never goes above 1. For example, a sound at an intensity of 1.0 would be multiplied by (19/20) to get 0.95. Addition of 0.05 is needed to get the value back to 1, as we don't want it to change in this case. A sound at an intensity of 0, (e.g. left ear when sound is coming directly from the right) will also be multiplied by (19/20) to get 0. We want it at the minimum level of 0.05, which is 0 + 0.05.

I think we should keep it as is in addition to blooming, because the volume on the opposite ear should not reach 0.

Reply:
okay. This kind of explanation should be added into the code as a comment in order to facilitate future programmers understanding this part :)